docs/DeveloperGuide.md

```diff
@@ -1,34 +1,503 @@
+This section describes the details of how each feature is implemented.
+
+### Add a calendar item feature
+This feature is facilitated by `AddCommand`.
+The following sequence diagram shows how the `execute()` operation works:<br/>
+<img src="https://raw.githubusercontent.com/AY2021S1-CS2113T-T12-2/tp/master/images/add_command_SD.JPG" alt="" width="750"/><br/>
+
+Note: A self-call to a specific method to add the new `CalendarItem` to the `CalendarList` is done based on the `CalendarItem` being added.  
+
+`CalendarItems` with deviations to sequence diagram above: 
+1. All `CalendarItems` with a date or time attribute calls the `TimeParser` class to interpret the date and time input of the user. 
+2. `Lecture`, `Lab`, `Tutorial`, or `Exam` type does a self-call for an additional check using the function `isValid(command)`. It ensures the module code included in the item is valid.
+3. `Lecture`, `Lab`, or `Tutorial`can be added as recurring items (multiple addition of the same class).   
+
+### Mark a task as done feature
+This feature is facilitated by `DoneCommand`.
+The following sequence diagram show how the `execute()` operation works:<br/>
+<img src="https://raw.githubusercontent.com/AY2021S1-CS2113T-T12-2/tp/master/images/done_command_SD.JPG" alt="" width="750"/><br/>
+Note: The command first extracts the task number from the user input prior to `convertTaskNumberToCalendarNumber`. 
+There will be a check in the function `markTaskAsDone(calendarNumber)` to ensure that the calendar item being marked as done is a task. 
+
+
+### Additional information of an event feature
+#### Add additional information
+This feature is facilitated by `AddInfoCommand`.
+The following sequence diagram show how the `execute()` operation of `AddInfoCommand` works:<br/>
+<img src="https://raw.githubusercontent.com/AY2021S1-CS2113T-T12-2/tp/master/images/addInfoCommand_SD.JPG" alt="" width="750"/><br/>
+
+#### View additional information
+This feature is facilitated by `ViewInfoCommand`.
+
+The following sequence diagram show how the `execute()` operation of `ViewInfoCommand` works:<br/>
+<img src="https://raw.githubusercontent.com/AY2021S1-CS2113T-T12-2/tp/master/images/viewInfoCommand_SD.JPG" alt="" width="750"/><br/>
+
+#### Delete additional information
+This feature is facilitated by `DeleteInfoCommand`.
+
+The following sequence diagram show how the `execute()` operation of `DeleteInfoCommand` works:<br/>
+<img src="https://raw.githubusercontent.com/AY2021S1-CS2113T-T12-2/tp/master/images/deleteInfoCommand_SD.JPG" alt="" width="750"/><br/>
+
+Note: The commands first extracts the event number from the user input prior to `convertEventNumberToCalendarNumber`. 
+
+### Delete a calendar item feature
+This feature is facilitated by `DeleteCommand`.
+The following sequence diagram show how the `execute()` operation works:<br/>
+<img src="https://raw.githubusercontent.com/AY2021S1-CS2113T-T12-2/tp/master/images/deleteCommand_SD.JPG" alt="" width="750"/><br/>
```

Comment:
![image](https://user-images.githubusercontent.com/18654251/97659815-eb565c00-1aaa-11eb-8a66-caf54dd12fd2.png)
Perhaps it would be more accurate if remove method call to the Task object be linked to an activation bar instead of directly to the object termination. The object can then be terminated soon after.

Also, it seems that the activation bar for Calendarlist is being overlapped by the objects lifeline.